src/index.cpp

```diff
@@ -1775,148 +1715,31 @@ std::unordered_map<std::string, LabelT> Index<T, TagT, LabelT>::load_label_map(c
     return string_to_int_mp;
 }
 
-template <typename T, typename TagT, typename LabelT>
-LabelT Index<T, TagT, LabelT>::get_converted_label(const std::string &raw_label)
-{
-    if (_label_map.find(raw_label) != _label_map.end())
-    {
-        return _label_map[raw_label];
-    }
-    if (_use_universal_label)
-    {
-        return _universal_label;
-    }
-    std::stringstream stream;
-    stream << "Unable to find label in the Label Map";
-    diskann::cerr << stream.str() << std::endl;
-    throw diskann::ANNException(stream.str(), -1, __FUNCSIG__, __FILE__, __LINE__);
-}
-
 template <typename T, typename TagT, typename LabelT>
 void Index<T, TagT, LabelT>::parse_label_file(const std::string &label_file, size_t &num_points)
```

Comment:
we should remove index::parse_label_file, and call load_labels in the calling function..  _filter_store->populate_labels can be called in build_filtered_index function, or whereever the label file is given in the build pipeline?